 this is a python exception
